app/models/good_job/job.rb

```diff
@@ -96,6 +96,33 @@ class Job < BaseRecord
     # @return [ActiveRecord::Relation]
     scope :only_scheduled, -> { where(arel_table['scheduled_at'].lteq(bind_value('scheduled_at', DateTime.current, ActiveRecord::Type::DateTime))).or(where(scheduled_at: nil)) }
 
+    # Exclude jobs that are paused via queue_name or job_class.
+    # Only applies when enable_pause configuration is true.
+    # @!method exclude_paused
+    # @!scope class
+    # @return [ActiveRecord::Relation]
+    scope :exclude_paused, lambda {
+      return all unless GoodJob.configuration.enable_pauses
+
+      paused_query = GoodJob::Setting.where(key: GoodJob::Setting::PAUSES)
+      paused_queues_query = paused_query.select("jsonb_array_elements_text(value->'queues')")
+      paused_job_classes_query = paused_query.select("jsonb_array_elements_text(value->'job_classes')")
+      paused_labels_query = paused_query.select("jsonb_array_elements_text(value->'labels')")
+
+      where.not(queue_name: paused_queues_query)
+           .where.not(job_class: paused_job_classes_query)
+           .where(
+             Arel::Nodes::Not.new(
+               Arel::Nodes::NamedFunction.new(
+                 "COALESCE", [
+                   Arel::Nodes::InfixOperation.new('&&', arel_table['labels'], Arel::Nodes::NamedFunction.new('ARRAY', [paused_labels_query.arel])),
+                   Arel::Nodes::SqlLiteral.new('FALSE'),
+                 ]
+               )
+             )
+           )
+    }
+
```

Comment:
THis is the hot query, now with labels, which makes it slightly more complicated.